### PR TITLE
Squirrel Search Algorithm implementation try

### PR DESCRIPTION
**Pull request by altaregos**
### Summary

Squirrel Search Algorithm implementation under ssa.py. The code does not have any bugs but the algorithms accuracy needs to be checked. 
Open Issue #338

### CODE REVIEW

**Comment by firefly-cpp** [comment]
Thank you @altaregos for your implementation.

I ask @zStupan to be reviewer of this implementation.

**Comment by firefly-cpp** [comment]
@altaregos, please add at least one test file as well as one example.  

See examples:

[DE tests](https://github.com/NiaOrg/NiaPy/blob/master/niapy/tests/test_de.py)

[DE example](https://github.com/NiaOrg/NiaPy/blob/master/examples/run_de.py)

File: niapy/algorithms/other/ssa.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            elif population_fitness[i]==beta_b_fitness:
+                a2=beta_b+0.8*1.9*(alpha-beta_b)
+                population[i] = task.repair(a2, rng=self.rng)
+            elif population_fitness[i]==beta_c_fitness:
+                a3=beta_c+0.8*1.9*(alpha-beta_c)
+                population[i] = task.repair(a3, rng=self.rng)
+            else:
+                k1 = random.randint(0, 1)
+                k2 = random.randint(0, 1)
+                k3 = random.randint(0, 1)
```
Comment (zStupan) [inline niapy/algorithms/other/ssa.py:193]: use self.integers(2) instead of random.randint.

**Review by zStupan — COMMENTED** [review]
This implementation appears to be missing some things from the [pseudocode](https://www.hindawi.com/journals/complexity/2019/6291968/alg1/), like the season monitoring condition. Also lower_limit, upper_limit and dimension parameters are never used, so they can be removed. It would also be nice to have the option to set the algorithm parameters in the constructor (I assume they're hard coded now).

Here's a [link](https://www.hindawi.com/journals/complexity/2019/6291968/) to an article which also contains the pseudocode and the description of the basic SSA, since the original article is behind a paywall.

EDIT: There also appear to be some [linter errors](https://github.com/NiaOrg/NiaPy/runs/4417825817?check_suite_focus=true). Please fix those. You can also check locally using the `make check` command.

**Pull request closed without merging**